crates/worker/src/partition/options.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use serde_with::serde_as;
+use std::time::Duration;
+
+/// Partition processor options
+#[serde_as]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, derive_builder::Builder)]
+#[cfg_attr(feature = "options_schema", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "options_schema", schemars(rename = "TimerOptions", default))]
+#[builder(default)]
+pub struct Options {
+    /// # Maximum batch size
+    ///
+    /// Specifies the maximum batch size of commands that the partition processor applies w/o
+    /// checking for signals from its actuators. The lower the value, the more disk writes are
+    /// being performed which reduces the overall throughput at the gain of potentially faster
+    /// invocation latencies.
+    pub max_batch_size: usize,
+
+    /// # Maximum batch duration
+    ///
+    /// The maximum duration the partition processor can spend on batching commands before
+    /// checking for signals from its actuators. The larger the value, the fewer disk writes
+    /// are being performed which can improve overall throughput at the cost of potentially
+    /// longer invocation latencies.
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    #[cfg_attr(feature = "options_schema", schemars(with = "String"))]
+    pub max_batch_duration: humantime::Duration,
```

Comment:
1. I'd suggest we only use a single option rather than two. Tuning both is going to be hard in practice. If tail-latency is most important then we might just keep the duration and remove the size limit.

Reply:
True, the more knobs there are the harder it is to find the right configuration. Starting with the max batch duration is probably a reasonable start.